Draft (#3)
lsst-dm/dmtn-276: DMTN-276.tex

% ADD CONTENT HERE
% You can also use the \input command to include several content files.

\section{Electro-optical Measured Persistence}
% \begin{figure*}[!htp]
%   \centering
%   \includegraphics[width=0.95\textwidth, angle=0]{Run_5_persistence_ex.png}
%   \caption{
%   Example of peristence in Run 5 data. 
%   This figure shows the initial flash on the left most image (dataId shown as the title) and the subsquent darks images going from left to right.
%   This particular flash has one of the three spots inbetween two amplifiers and you can see the trail starting at the spots and going all the way to the readout of each amplifier.
%   }\label{fig:ex_persistence_Run5}
% \end{figure*}

\subsection{Definition and Run 5 Measurements}

\begin{figure*}[!htp]
  \centering

>   \includegraphics[width=0.95\textwidth, angle=0]{Run_5_persistence_ex.png}

Author: This ends up being too small to see in the PDF.

  \caption{
  Example of peristence in Run 5 data. 
  This figure shows the initial flash on the left most image (dataId shown as the title) and the subsquent darks images going from left to right.
  This particular flash has one of the three spots inbetween two amplifiers and you can see the trail starting at the spots and going all the way to the readout of each amplifier.
  }\label{fig:ex_persistence_Run5}
\end{figure*}

Persistence is a sensor level effect where charge gets trapped from one image and `persists' into the next at some fraction of the original charge. 
Figure \ref{fig:ex_persistence_Run5} shows an example of this from Run 5 data where three saturated spots persist into subsequent dark images. 
Persistence for LSSTCam was first found in Run 5 of electro-optical (EO) testing at SLAC from crosstalk measurements. 
Further investigation found that this effect only affected E2V sensors
% This persistence was found to have an average signal after the flash of 6 ADU and had a decay constant of 37 seconds. 

> and that the persistence was only mitigated via integration time as no persistence was shown in biases. 

Author: I'm confused about what "mitigated via integration time" means and what the comment about biases. No persistence was seen in biases because they're not saturated? If you take a bias right after a bright star, doesn't it show persistence?

Reviewer: So I changed it to try and explain but just in case it is still confusing, I will try and do it here. Essentially, there were some tests done to measure and mitigate the persistence and one of those was taking biases after the bright flash and seeing if that helped with persistence. It didn't as after the biases, there persistence was still visible in a dark image.

> One of the troublesome features of the persistence in the E2V sensors is the persistence trail. 

Author: This could be clearer if you describe it as part of the readout process, e.g. "In addition to the persistence from the bright source, the pixels through which the source is transferred during readout also show persistence, leaving a trail from the star to the edge of the amplifier"

Not only do the saturated pixels have an increased signal, but so do all the pixels leading to the end of the amplifier in the parallel direction.

\subsection{Run 6 Measurements}
Following this discovery in Run 5, we tested and characterized the persistence further in Run 6a and 6b. 
This included using both a flat illumnination to induce persistence as well as utilizing a pinhole filter to create saturated spots to measure the persistence.

\subsubsection{Flat Induced Persistence}
\begin{figure*}[!htp]
  \centering
  \includegraphics[width=0.45\textwidth, angle=0]{Run_13392_Persistence.png}
  \includegraphics[width=0.45\textwidth, angle=0]{Persistence_Flat_Example.png}
  \caption{
  Full focal plane images of the counts in the first dark image after a bright exposure for run 13392.
  The left shows the amplifier level average for all science sensors of the subsequent dark.
  The right shows a binned image of the dark image from RubinTV\@.
  % This particular flash has one of the three spots inbetween two amplifiers and you can see the trail starting at the spots and going all the way to the readout of each amplifier.

>   }\label{fig:Flat_Induced}

Author: Color bar labels are too small to read.

\end{figure*}

Flat illuminated persistence measurements were taken during B and C protocol runs.
The procedure for these measurements were to take a incredibly bright (3--4 times full well) flat followed by 15 second dark images.
These images are then run through both \texttt{$cp\_pipe$} and \texttt{$eo\_pipe$}.
This procedure is designed to induce the persistence, determine the level and then track its decay.

> Unfortunately this method does not allow us to measure the trail caused by the the persistence as all the pixels are saturated.

Author: "the the"

Reviewer: Fixed

\begin{figure*}[!htp]
  \centering
  \includegraphics[width=0.45\textwidth, angle=0]{Run_13392_Persistence.png}
  \includegraphics[width=0.45\textwidth, angle=0]{Run_13410_Persistence_Lower.png}
  \caption{
  Full Focal plane images with amplifier level resoltuion comparing a subsequent dark image with nomial parallel clock swing voltages (left; Run 13392) 
  and one with a slightly higher parallel clock swing voltage (right; Run 13410).
  % This particular flash has one of the three spots inbetween two amplifiers and you can see the trail starting at the spots and going all the way to the readout of each amplifier.

>   }\label{fig:Flat_Induced_Comparison}

Author: Same colorbar label issue as above. If there's a way to get rid of some of the margins in the image, that would help make the focal planes bigger and easier to see.

\end{figure*}

> One of the concerning discoveries with these measurements were the level of persistence was reaching 10 ADU on a per amp basis.

Author: "per amp basis", is this saying that persistence differs between amps and can reach as high as 10 ADU? Is the concern the per-amp aspect or the amplitude?

Reviewer: Changed the text to highlight that the concern is with the 10 ADU but it is also interesting how it changes from detector to detector and even amp to amp in a detector.

Figure \ref{fig:Flat_Induced} shows an example of a focal plane plot displaying the per amp value of the first dark image after the flash as well as the actual dark image.

> One of the ways to lower or remove the persistence effect is to adjust the parallel swing voltages of the detector.

Author: This seems to switch topics to mitigation mid-paragraph.

Reviewer: Changed to two paragraphs

This was tested and Figure \ref{fig:Flat_Induced_Comparison} comapares the per amp plot between the nominal voltages and the slightly higher parallel clock swing voltages (-9.3 vs -9.0 V).
This change did lower the persistence level but much less than what was expected (only $10\%$).

\subsubsection{Spot Induced Persistence}
During Run 6b, we utilized the pinhole filter to measure the persistence of the E2V sensors.
The pinhole filter is positioned so that a small area of light hits one amplifier on each of the middle detectors on a raft.
This means that we can only measure the persistence of 13 e2V detectors but we are able to get measurements across the focal plane.
For these tests, we started with flashes from around 20 ke-/pixel below full well (~100 ke-/pixel) and ramped to twice full well (200 ke-/pixel).

> After each flash, we take 10, 15 second dark images which we use to measure the persistence.

Author: take 10 images, each 15 seconds long?

Reviewer: Yes, I changed to the text to clear that up 'After each flash, we take 10 dark images with 15 second integration times which we use to measure the persistence.'

These images are then overscan and bias corrected.
Figure \ref{fig:Run6_example} shows an example of one of the saturated pixels (spot) and the subsquent dark image with persistence.

\begin{figure*}[!htp]
  \centering
  \includegraphics[width=0.85\textwidth, angle=0]{Run6_ex.png}
  \caption{
  Similar to Figure \ref{fig:ex_persistence_Run5} but with the pinhole filter. 

>   This shows the initail flash and the spot (left) and the dark imgaes that followed.

Author: ```suggestion
  This shows the initial flash and the spot (left) and the dark images that followed.
```

  }\label{fig:Run6_example}
\end{figure*}

Using these images, we are able to measure the persistence induced by the saturated pixels as well as the trail that follows.
We then model the persistence decay using this function:
\begin{equation*}

>   Counts= A * \exp(t/\tau)+c

Author: \textrm{Counts}

\end{equation*}

> where \textit{Counts} are the average number of counts at the location of the spot in the following dark image, \textit{t} is the integration time since the flash, \textit{$\tau$} is the decay constant and \textit{A} and \textit{c} are constants.

Author: Is t really the integration time and not the wallclock time? Do you know that the difference matters? Also just in the interesting getting the specifics written down, is t at the start of the dark integration, midpoint, or end?

Reviewer: It is the wallclock time, as I am just taking the MJD from the images (I think this is set as the beginning of integration). I will change the text to say 'time since flash, 15 seconds of which are with the CCDs integrating'

Figure \ref{fig:Run6_decay} shows an example of the decay of the both the spot and the trail and the corresponding model fit for one detector.

\begin{figure*}[!ht]
  \centering

>   \includegraphics[width=0.95\textwidth, angle=0]{Run_6_decay.png}

Author: typo in the left panel legend, "15.2 seconds"

  \caption{
  The decay of the persistence with consecutive dark images. 
  The blue points are the measured average counts of the spot (left) and trail (right).
  The red line shows the model fit with the parameters shown in the upper right.
  }\label{fig:Run6_decay}
\end{figure*}

> Figure \ref{fig:Pers_Flash_Tau} shows the values for \textit{A} and \textit{$\tau$} for all detectors and all exposures that showed persistence. 

Author: is "all detectors that showed persistence" the same as "all measured e2v sensors"?

Reviewer: I tried to change the text to reflect this but it is weird as those two statements are not the same. The differences between detectors in persistence level is also seen in how easy/hard it is to induce persistence. So like some detectors have 12/20 persistence inducing exposures while others have only 4.

Reviewer: Sorry, I totally misread the comment. I changed it to all measured e2v detectors.

This figure shows that while \textit{$\tau$} is roughly the same for all images (between 10--20 seconds), the amplitude of the effect changes.

> There also is a relationship between flashtime/the brightest of the spot and the persistence measured signal.

Author: I'd phrase it as "between the brightness of the spot (controlled by the flash duration in these experiments) and the measured persistence amplitude."

\begin{figure*}[!ht]
  \centering
  \includegraphics[width=0.95\textwidth, angle=0]{Persistence_Flash_Tau.png}
  \caption{

>   The relationship between flash time (a proxy for a bright source) and amplitude (left) and tau (right) for the spot. 

Author: Is flash duration a less ambiguous way to describe this? I think I'd say it's a proxy for the brightness of the saturated source. Each point here is a single sensor from a single image, or something else?

Reviewer: Changed it to 'duration of the LED flash' and 'a proxy for the brightness of a saturated source'. I also added 'Each point represents the measured tau and amplitude values of a single sensor from a single exposure that displayed persistence.' to help with the ambiguity of what is being plotted.

  The amplitude appears to be dependent on the flash time, while tau is invariant and is consistently between 10--20 seconds. 
  }\label{fig:Pers_Flash_Tau}
\end{figure*}

\section{Effect of Persistence on DC2 Images}

> Though studies are currently being done to minimize or elminate persistence, both through hardware and processing, one of the next steps is to see how much persistence will affect LSST given its current rought characterization.

Author: "affect LSST given its" -> something like "affect LSST photometric measurements", leave off rough characterization.

Reviewer: Changed, I think the goal originally was to say that there is work being done to better characterize the persistence, but we will use these rough assumptions (looking back, I don't think that's necessary).

> To study these effects, we used LSST DC2 data and the rough characterization that all persistence has a \textit{$\tau$} of 15 seconds and all saturated pixels have a amplitude of 5 ADU\@.

Author: DESC DC2 data is probably the better way to describe it. Maybe "a persistence amplitude" would avoid ambiguity?

Reviewer: Changed the sentence to 'To study these effects, we used DESC DC2 data and assume that the persistence will affect all saturated pixels with the characteristics of a \textit{$\tau$} of 15 seconds and an amplitude of 5 ADU.'

> These numbers are similar to what we see with the pinhole data, which we feel will better represent on-sky data.

Author: "similar to" -> "based on", I think?

Reviewer: Changed to 'derived from the pinhole data' to make it sound bit more professional.

> As most of the characterization has been with the spot and not the trail, we use 5 different schema to determine the impact of the trail: assume the trail carries with it 100\%, 75\%, 50\%, 25\%, and 0\% of the spot persistence level.

Author: I think I'd call these models instead of schema.

Reviewer: Changed

We only used one of the e2V detectors at random (detector number 88/R21-S21) for this analysis and then assume that this detector is representative of the other e2V detectors.

> We also only utilize single visit images to mimic what the effect would be as images would be coming from the observatory.

Author: This means single-visit instead of coadds? Or single-visit instead of two-snaps?

Reviewer: Trying to say single visit instead of coadds. Though do you think it would be worth it to say single 30 second exposures? From what I saw, that's what all DC2 is but it might be worth it in case readers think that they are snaps. Right now I changed it to '... single-visit images with 30 second exposures ...'

> \subsection{Persistence Affected Pixels}

Author: Maybe "Affected pixel counts" or similar? Affected can read like a verb here, which I think was not your intention.

Reviewer: Changed `subsection` title to Affected Pixel Counts

> One of the first studies we did was to look at the number of affected pixels due to persistence during a simulated night of observing.

Author: ```suggestion
One of the first studies we did was to look at the number of pixels affected by persistence during a simulated night of observing.
```

Reviewer: Changed

> To do this, we utilized the DC2 dataset located at the Google data facility.

Author: ```suggestion
To do this, we utilized the DC2 dataset
```

Reviewer: Changed

> Taking 1000 images, we simulated the persistence by having a running list of footprints with persistence.

Author: Footprints is used by pipelines for a very specific purpose, so I'd try to call this something different. 

Separately, I think it's worth describing the process you used to generate the simulated images first, and then second describing the count of affected pixels and the effects on fluxes. (Even if in code you did these separately, they're both following the same idea of taking the image sequence and calculating the persistence effects)

Reviewer: So for this part, I am using a list of `footprintSets`, should I texttt that to show that or still change it to something else?

I will add in more detail to the process for both. Though for the pixel counting, I just used `calexp` images to find saturated pixels, and add it to persistence mask (including the trail). And then as you mention, something different for the flux effects (which I will go into in the next section).

If the persisence dropped below 1 ADU, the footprints were taken off the list, as this is starting to reach the level of the bias noise.
Table \ref{tab:persis_affected}shows the average number of affected pixels in each of the schema.

\begin{table}[h!]
\centering
\begin{tabular}{|lcccc|}
\hline
\% of Spot in Trail & >3 ADU ($10^{-4}$) & >1 ADU ($10^{-4}$) & >0.1 ADU ($10^{-4}$) & >0.01 ADU ($10^{-4}$) \\
\hline
100 & 5.85 & 5.85 & 11.7 & 23.4 \\
75 & 5.85 & 5.85 & 11.7 & 17.5 \\
50 & 0.0216 & 5.85 & 11.7 & 17.5 \\
25 & 0.0216 & 5.85 & 11.7 & 17.5 \\
0 & 0.0216 & 3.91 & 7.81 & 11.7 \\
% Dark &
\hline
\end{tabular}\label{tab:persis_affected}
\caption{Table showing the average about of pixels in persistence affected regions above ceratin ADU levels. 
The numbers are then normalized to fraction of total CCD pixels affected.
We compare with different trail levels.
%  taken from a recent EO testing Opsim run.
}  
\end{table}

> \subsection{Persistence Affected Objects}

Author: "Effects on Objects" maybe?

Reviewer: Changed to Persistence Induced Flux Effects on Objects

> Another possible metric that could be impacted by the persistence are the object measurements.

Author: These aren't really a metric, they're the main science product.

Reviewer: Changed

For this study, we created a new dataset type called \texttt{$raw\_modified$}. 
This dataset type takes the \texttt{$raw$} file and adds the persistence model as described above utilizing the previous images.
For this study, we utilized a small subset of the DC2 dataset (68 images) and the latest reprocessing using the latest verison of the Data Release Pipeline (DRP).
We then run the \texttt{$raw\_modified$} images through the \texttt{$isr$}, \texttt{$characterizeImage$}, and \texttt{$calibrate$} steps of the DRP\@. 

Figure \ref{fig:ex_persistence} shows an example image of an object comparing the two \texttt{$calexp$} images from \texttt{$raw$} and \texttt{$raw\_modified$}.

> Using these 68 images, we measured the differences in magnitude of objects centered in the persistence affected regions.

Author: "persistence-affected" throughout.

Reviewer: Changed

Figure \ref{fig:affected_objects_all} shows the difference in magnitude using both the an aperture flux and PSF flux.

\begin{figure*}[!htp]
  \centering
  \includegraphics[width=0.95\textwidth, angle=0]{Obj_pers.png}
  \caption{

>   A zoomed in object that shows the persistence affected image (left), the original image (middle) 

Author: I would drop this image, there's really not much that you can see by-eye.

Reviewer: I wanted to keep this image just to show that the persistence is not very visible by eye, as that was a major comment by the camera folks when first talking about this (that its effect is swallowed up by the noise). I don't know if that is as effective to write that out though.

  and the subtraction between the two to highlight the persistence (right).
  }\label{fig:ex_persistence}
\end{figure*}

\begin{figure*}[!htp]
  \centering
  \includegraphics[width=0.95\textwidth, angle=0]{Total_Objects_affected.png}
  \caption{

>   The changes of magnitude of objects in persistence affected regions for all five trail regimes. 

Author: This is really the key plot, so it's worth spending extra effort on it. My feeling is that there are more panels than you need, and that makes the individual details hard to read. Maybe have a single 100%-trail plot, and then separately a multi-panel comparison? Few details:
- typo in the legend
- All the axes labels need to be bigger
- Fraction of affected sources instead of total count. 
- The points are kind of too big to see much detail? Maybe making them smaller would help? Or adding a "mean mag difference" line? I'm not exactly sure, but it could use some tweaking. 
- PSF and aperture don't really show well together on the same plot. Maybe that would get better if the points were smaller, otherwise you might just have a supporting figure that shows that both are pretty equally affected.

I'm not sure what is the best way to convey whether the points are showing a tiny subset of the sources in the images or a large fraction. Switching to a fraction in the label will help, but there might be other options like including all sources, but switching to a binned 2d histogram so people can see both the unaffected and affected sources on the same plot? I wouldn't use this latter idea as a replacement for the figure you already have, maybe just as a supplement if it works out.

Reviewer: Working on the plot. Right now, I have the fraction of sources affected O(1e-4) underneath the Number of Affect Sources.

Then I think it is worth it to make the points smaller and maybe change the alpha to put them in the background and include a larger point of the average for certain bins (every 0.5 mag or something). I just also want to include the spread to show worst case scenarios.
